src/components/side_nav/side_nav_item.tsx

```diff
@@ -16,16 +16,38 @@ type ItemProps = CommonProps & {
   children: ReactNode;
 };
 
-export type EuiSideNavItemProps = ItemProps & {
+interface SideNavItemProps {
   isOpen?: boolean;
   isSelected?: boolean;
   isParent?: boolean;
   icon?: ReactElement;
   items?: ReactNode;
   depth?: number;
-  renderItem?: (props: ItemProps) => JSX.Element;
+}
+
+type ExcludeEuiSideNavItemProps<T> = Pick<
+  T,
+  Exclude<keyof T, keyof SideNavItemProps | 'renderItem'>
+>;
+type OmitEuiSideNavItemProps<T> = {
+  [K in keyof ExcludeEuiSideNavItemProps<T>]: T[K]
 };
 
+interface GuaranteedRenderItemProps {
+  href: string | undefined;
```

Comment:
```suggestion
  href?: string;
```

Reply:
Had a reason for doing this, but looks like it doesn't exist anymore.